exonum/src/runtime/dispatcher.rs

```diff
@@ -137,7 +134,12 @@ impl RuntimeEnvironment for Dispatcher {
         }
 
         if let Some(runtime) = self.runtimes.get(&runtime_id.unwrap()) {
-            runtime.execute(context, call_info, payload)
+            runtime.execute(context, call_info, payload)?;
+            // Executes pending dispatcher actions.
+            context
```

Comment:
Maybe add a check if there is any action so we won't create an empty `Vec` on each transaction execute?

Reply:
Empty vec doesn't allocate memory and thus `Vec::new` is very cheap.